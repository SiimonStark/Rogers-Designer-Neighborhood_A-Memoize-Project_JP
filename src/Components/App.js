import React, { Component } from 'react';
import Header from './Header';
import DisplayField from './DisplayField';
import PlayerCommands from './PlayerCommands';

import Prompts from './Prompts';

class App extends Component {
  constructor() {
    super()
    this.state = {
      playerInputs: [
        {gridInputs: []},
        {flexInputs: []}
      ]
    }
  }
  updateInputs = (field, userInput) => {
    let grid = this.state.playerInputs[0].gridInputs;
    let flex = this.state.playerInputs[1].flexInputs;
    
    grid[field] = userInput;
    flex[field] = userInput;

    console.log('grid', grid)
    console.log('flex', flex)
    this.setState({
      playerInputs: [
        {gridInputs: grid},
        {flexInputs: flex}
      ]
    })
  }

  render() {
    console.log('player Inputs: ', this.state.playerInputs)
    return (
      <div className="App">
        <Header />
        <DisplayField updateInputs={this.updateInputs} />
        <PlayerCommands updateInputs={this.updateInputs} />
        <Prompts updateInputs={this.updateInputs} />
      </div>
    );
  }
}

export default App;