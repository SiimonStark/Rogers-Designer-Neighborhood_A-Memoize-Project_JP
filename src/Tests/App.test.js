import React from 'react';
import ReactDOM from 'react-dom';
import App from '../Components/App';
import { shallow } from "enzyme";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

const defaultState = {
  playerInputs: [{ gridInputs: [] }, { flexInputs: [] }],
  promptAnswers: []
};

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should match screenshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    expect(wrapper.state()).toEqual(defaultState);
  });
})