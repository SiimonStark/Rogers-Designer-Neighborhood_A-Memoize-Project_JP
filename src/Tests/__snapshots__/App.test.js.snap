// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match screenshot 1`] = `
<div
  className="App"
>
  <Header />
  <DisplayField />
  <PlayerCommands
    updateInputs={[Function]}
  />
  <Prompts
    playerInputs={
      Array [
        Object {
          "gridInputs": Array [],
        },
        Object {
          "flexInputs": Array [],
        },
      ]
    }
    updatePromptAns={[Function]}
  />
</div>
`;
